File: core/src/SprCSharp/SprExport/tmp/vectorwrapper_HISpidarMotorDesc.cpp
    __declspec(dllexport) HANDLE __cdecl Spr_vector_new_HISpidarMotorDesc() {
        HANDLE _ptr = NULL;
        try { _ptr = new vector<HISpidarMotorDesc>; }
        catch (SEH_Exception e) { e.raise_managed_exception("SprExport.dll"); }
        return _ptr;
    }
    __declspec(dllexport) void __cdecl Spr_vector_delete_HISpidarMotorDesc(HANDLE ptr) {
        if (ptr == NULL) return;
        vector<HISpidarMotorDesc>* vec = (vector<HISpidarMotorDesc>*) ptr;
        vector<HISpidarMotorDesc>().swap(*vec);
    }
    __declspec(dllexport) HANDLE __cdecl Spr_vector_get_HISpidarMotorDesc(HANDLE ptr, int index) {
        HANDLE _ptr = NULL;
        try {
            vector<HISpidarMotorDesc>* vecptr = (vector<HISpidarMotorDesc>*) ptr;
            _ptr = (HANDLE) &(*vecptr)[index];
        }
        catch (SEH_Exception e) { e.raise_managed_exception("SprExport.dll"); }
        return _ptr;
    }
    __declspec(dllexport) void __cdecl Spr_vector_set_HISpidarMotorDesc(HANDLE ptr, int index, HANDLE value) {
        try {
            vector<HISpidarMotorDesc>* vecptr = (vector<HISpidarMotorDesc>*) ptr;
            (*vecptr)[index] = *((HISpidarMotorDesc*) value);
        }
        catch (SEH_Exception e) { e.raise_managed_exception("SprExport.dll"); }
    }
    __declspec(dllexport) int __cdecl Spr_vector_size_HISpidarMotorDesc(HANDLE ptr) {
        int _val = 0;
        try {
            vector<HISpidarMotorDesc>* vecptr = (vector<HISpidarMotorDesc>*) ptr;
            _val = (int) (*vecptr).size();
        }
        catch (SEH_Exception e) { e.raise_managed_exception("SprExport.dll"); }
        return _val;
    }
    __declspec(dllexport) void __cdecl Spr_vector_push_back_HISpidarMotorDesc(HANDLE ptr, HANDLE value) {
        try {
            vector<HISpidarMotorDesc>* vecptr = (vector<HISpidarMotorDesc>*) ptr;
            (*vecptr).push_back(*((HISpidarMotorDesc*) value));
        }
        catch (SEH_Exception e) { e.raise_managed_exception("SprExport.dll"); }
    }
    __declspec(dllexport) void __cdecl Spr_vector_clear_HISpidarMotorDesc(HANDLE ptr) {
        try {
            vector<HISpidarMotorDesc>* vecptr = (vector<HISpidarMotorDesc>*) ptr;
            (*vecptr).clear();
        }
        catch (SEH_Exception e) { e.raise_managed_exception("SprExport.dll"); }
    }

File: core/src/SprCSharp/SprCSharp/tmp/vectorwrapper_HISpidarMotorDesc.cs
    public class vectorwrapper_HISpidarMotorDesc : vectorwrapper {
        public vectorwrapper_HISpidarMotorDesc() {
            _this = SprExport.Spr_vector_new_HISpidarMotorDesc();
            _flag = true;
        }
        public vectorwrapper_HISpidarMotorDesc(IntPtr ptr) : base(ptr) {}
        public vectorwrapper_HISpidarMotorDesc(IntPtr ptr, bool flag) : base(ptr, flag) {}
        ~vectorwrapper_HISpidarMotorDesc() {
            if (_flag) { SprExport.Spr_vector_delete_HISpidarMotorDesc(_this); _flag = false; }
        }
        public HISpidarMotorDesc this[int index] {
            get { return new HISpidarMotorDesc(SprExport.Spr_vector_get_HISpidarMotorDesc(_this, index)); }
            set { SprExport.Spr_vector_set_HISpidarMotorDesc(_this, index, value); }
        }
        public int size() { return (int) SprExport.Spr_vector_size_HISpidarMotorDesc(_this); }
        public void push_back(HISpidarMotorDesc value) { SprExport.Spr_vector_push_back_HISpidarMotorDesc(_this, value); }
        public void clear() { SprExport.Spr_vector_clear_HISpidarMotorDesc(_this); }
        public static implicit operator List<HISpidarMotorDesc>(vectorwrapper_HISpidarMotorDesc m) {
            List<HISpidarMotorDesc> r = new List<HISpidarMotorDesc>();
            int size = SprExport.Spr_vector_size_HISpidarMotorDesc(m._this);
            for (int i = 0; i < size; i++) {
                IntPtr member = SprExport.Spr_vector_get_HISpidarMotorDesc(m._this, i);
                r.Add(new HISpidarMotorDesc(member));
            }
            return r;
        }
        public static implicit operator vectorwrapper_HISpidarMotorDesc(List<HISpidarMotorDesc> r) {
            IntPtr ptr = SprExport.Spr_vector_new_HISpidarMotorDesc();
            vectorwrapper_HISpidarMotorDesc m = new vectorwrapper_HISpidarMotorDesc(ptr, true);
            for (int i = 0; i < (int) r.Count; i++) {
                SprExport.Spr_vector_push_back_HISpidarMotorDesc(ptr, r[i]);
            }
            return m;
        }
        public static implicit operator HISpidarMotorDesc[](vectorwrapper_HISpidarMotorDesc m) {
            int size = SprExport.Spr_vector_size_HISpidarMotorDesc(m._this);
            HISpidarMotorDesc[] r = new HISpidarMotorDesc[size];
            for (int i = 0; i < size; i++) {
                 IntPtr ptr = SprExport.Spr_vector_get_HISpidarMotorDesc(m._this, i);
                 r[i] = new HISpidarMotorDesc(ptr);
            }
            return r;
        }
        public static implicit operator vectorwrapper_HISpidarMotorDesc(HISpidarMotorDesc[] r) {
            IntPtr ptr = SprExport.Spr_vector_new_HISpidarMotorDesc();
            vectorwrapper_HISpidarMotorDesc m = new vectorwrapper_HISpidarMotorDesc(ptr, true);
            for (int i = 0; i < r.Length; i++) {
                SprExport.Spr_vector_push_back_HISpidarMotorDesc(m._this, r[i]);
            }
            return m;
        }
    }
